udata_front/theme/gouvfr/assets/js/components/MarkAsNoSpam/MarkAsNoSpam.vue

```diff
@@ -0,0 +1,37 @@
+<template>
+  <button
+    class="fr-btn fr-btn--sm fr-btn--icon-right fr-icon-feedback-fill"
+    @click="markAsNoSpam"
+    v-if="show"
+  >
+    {{$t('Mark as no spam')}}
+  </button>
+</template>
+<script setup lang="ts">
+import { computed, defineProps } from "vue";
+import { user } from "../../config";
+import { api } from "../../plugins/api";
+
+const props = defineProps<{
+  url: string,
+  spam: {
+    status: string
+  }
+}>();
+
+const markAsNoSpam = () => {
+  return api
+    .delete(props.url)
+    .then(() => {
+      window.location.reload();
+    });
+};
+
+const show = computed(() => props.spam
```

Comment:
You can remove both `props.spam` and `props.spam.status` conditions based on the props type

Reply:
Set the type as maybe undefined because some old objects may not have these properties…